### PR TITLE
Support additional {details} format for pay response

### PR DESCRIPTION
**Pull request by dvoytenko**
/to @kaipakartik

### CODE REVIEW

File: src/runtime/pay-flow.js [hunk 1]
```diff
@@ -173,7 +173,10 @@ export function parseSubscriptionResponse(data) {
     } else {
       // Assume it's a json object in the format:
       // `{integratorClientCallbackData: "..."}`.
-      const json = /** @type {!Object} */ (data);
```
Comment (kaipakartik) [inline src/runtime/pay-flow.js]: You weren't base64 decoding the data anywhere?

File: src/runtime/pay-flow.js [hunk 2]
```diff
@@ -166,26 +166,33 @@ export function validatePayResponse(port) {
...
+      const parsed = parseJson(raw);
+      swgData = parsed['swgCallbackData'];
+    }
   }
   if (!swgData) {
     throw new Error('unexpected payment response');
   }
+  raw = JSON.stringify(/** @type {!JsonObject} */ (swgData));
   return new SubscribeResponse(
       raw,
```
Comment (kaipakartik) [inline src/runtime/pay-flow.js]: Return the json object instead of Json.stringify?
It's fine to do this in another pull request if needed

File: src/runtime/pay-flow.js [hunk 3]
```diff
@@ -166,26 +166,33 @@ export function validatePayResponse(port) {
...
   }
-
-  let swgData = null;
-  if (raw) {
-    const parsed = parseJson(atob(raw));
-    swgData = parsed['swgCallbackData'];
+  if (raw && !swgData) {
+    raw = atob(raw);
+    if (raw) {
+      const parsed = parseJson(raw);
```
Comment (kaipakartik) [inline src/runtime/pay-flow.js]: Why not Json.Parse?

File: src/runtime/pay-flow.js (hunk 2 above)
Comment (dvoytenko) [reply src/runtime/pay-flow.js]: By current format, `raw` is a string to simplify posting it to the 3p site.

File: src/runtime/pay-flow.js (hunk 3 above)
Comment (dvoytenko) [reply src/runtime/pay-flow.js]: That's pretty much that - with some nuances around types.

Comment (dvoytenko) [reply src/runtime/pay-flow.js]: Also. That call is inlined into `JSON.parse()` in the binary.
